kanalony-storage-logic/src/main/scala/QueryParamsValidator.scala

```diff
@@ -0,0 +1,81 @@
+package kanalony.storage.logic
+
+import kanalony.storage.logic.queries.model.DimensionEqualityConstraint
+
+/**
+ * Created by elad.benedict on 2/11/2016.
+ */
+
+class QueryParamterException(message : String) extends IllegalArgumentException(message)
+
+case object QueryParamsValidator {
+
+  private def extractEqualityConstraintParam[T](dimension : Dimensions.Value, queryParams : QueryParams): List[T] = {
+    val dimDef = queryParams.dimensionDefinitions.find(_.dimension == dimension)
+    try {
+      dimDef match {
+        case Some(eqCons) => {
```

Comment:
Matching on Option[T] is not a good practice (see [this](http://stackoverflow.com/questions/8559537/where-does-the-flatmap-that-s-idiomatic-expression-in-scala-come-from/8561903#8561903) for example), if you can simply check:
`dimDef.isDefined`
I would match on the DimensionEqualityConstraint and throw QueryParamterException on _